### PR TITLE
rework eslint config

### PR DESCRIPTION
**Pull request by silverwind**
- use new explicit config that specifies each rule
- upgrade eslint to 7.1.0
- add new plugins with selected rules enabled
- fix discovered issues, remove global wipPrefixes

Rules are mostly based on [my config](https://github.com/silverwind/eslint-config-silverwind/blob/master/.eslintrc). Two of the plugins still don't specify eslint 7 compatibilty, but I think it's no issue, they do work fine.

### CODE REVIEW

File: templates/repo/issue/new_form.tmpl [hunk 1]
```diff
@@ -15,7 +15,7 @@
 					<div class="field">
 						<input name="title" id="issue_title" placeholder="{{.i18n.Tr "repo.milestones.title"}}" value="{{.title}}" tabindex="3" autofocus required maxlength="255">
 						{{if .PageIsComparePull}}
-							<div class="title_wip_desc">{{.i18n.Tr "repo.pulls.title_wip_desc" (index .PullRequestWorkInProgressPrefixes 0| Escape) | Safe}}</div>
+							<div class="title_wip_desc"{{if .PageIsComparePull}} data-wip-prefixes="{{Json .PullRequestWorkInProgressPrefixes}}"{{end}}>{{.i18n.Tr "repo.pulls.title_wip_desc" (index .PullRequestWorkInProgressPrefixes 0| Escape) | Safe}}</div>
```
Comment (zeripath) [inline templates/repo/issue/new_form.tmpl]: The if here seems superfluous

Comment (silverwind) [reply templates/repo/issue/new_form.tmpl]: Probably is, I just preserved it as it was present in the inline script below. You think it's fine to remove still?

Comment (silverwind) [reply templates/repo/issue/new_form.tmpl]: Removed it.

Comment (zeripath) [reply templates/repo/issue/new_form.tmpl]: Yeah it's already inside an {{if .PageIsComparePull}} one line above

Comment (silverwind) [reply templates/repo/issue/new_form.tmpl]: Ah, missed that.

File: templates/base/head.tmpl [hunk 2]
```diff
@@ -78,14 +78,14 @@
 				MaxTimeout: {{NotificationSettings.MaxTimeout}},
 				EventSourceUpdateTime: {{NotificationSettings.EventSourceUpdateTime}},
 			},
-      {{if .RequireTribute}}
+{{if .RequireTribute}}
```
Comment (zeripath) [inline templates/base/head.tmpl]: Has eslint forced you to move this to column 0?

That could be irritating

Comment (silverwind) [reply templates/base/head.tmpl]: It's not directly related here, I just noticed the `if` was intended with spaces but the `end` with tabs which was a bit distracting and it seemed better to just unintend that block. We can remove this change if necessary.

Comment (silverwind) [reply templates/base/head.tmpl]: Undid it.

Comment (zeripath) [reply templates/base/head.tmpl]: Oh I was just worried that eslint had made you change it!

**Comment by silverwind** [comment]
I can undo the config/dependency changes if requested.

There is a certain benefit of depending on a shared config because much less work is required to update the config, even thought we need to disable 20-something rules that are too opinionated in the airbnb config.

**Review by zeripath — APPROVED** [review]

**Review by gary-kim — APPROVED** [review]
:tada:

**Comment by silverwind** [comment]
Updated eslint and plugins again. `eslint-plugin-import` is now officially compatible with eslint 7 which only leaves `eslint-plugin-sonarjs` not officially compatible (but still workes). I'm still quite fond of `eslint-plugin-sonarjs` because it does provide a few useful rules.

**Comment by silverwind** [comment]
Rebased and fixed one more new violation (detected by `eslint-plugin-sonarjs` 😉) that came up.

**Review by lafriks — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping LG-TM

**Pull request merged**